lib/rex/version.rb

```diff
@@ -0,0 +1,12 @@
+
+class Rex::Version < Gem::Version
+
+  def initialize(version)
+    if version == nil
+      # Rubygems 4 is deprecating `nil` as a valid version number
+      # Currently it is the equivalent of a `0` so we set that here to keep the same functionality
+      version = 0
```

Comment:
I think this approach makes sense to me for now, it keeps us consistent with the current API. Long term we should decide how we want to the scenario of handle nil/empty strings. Nothing actionable for this PR though.